torchrl/modules/tensordict_module/common.py

```diff
@@ -217,22 +218,25 @@ def register_spec(self, safe, spec):
                     f"got more than one out_key for the TensorDictModule: {self.out_keys},\nbut only one spec. "
                     "Consider using a CompositeSpec object or no spec at all."
                 )
-            spec = CompositeSpec(**{self.out_keys[0]: spec})
+            spec = CompositeSpec({self.out_keys[0]: spec})
         elif spec is not None and isinstance(spec, CompositeSpec):
             if "_" in spec.keys() and spec["_"] is not None:
                 warnings.warn('got a spec with key "_": it will be ignored')
         elif spec is None:
             spec = CompositeSpec()
 
-        if set(spec.keys(True, True)) != set(self.out_keys):
+        # unravel_key_list(self.out_keys) can be removed once 473 is merged in tensordict
+        spec_keys = sorted(unravel_key_list(list(spec.keys(True, True))), key=str)
+        out_keys = sorted(unravel_key_list(self.out_keys), key=str)
+        if set(spec_keys) != set(out_keys):
```

Comment:
we can anticipate the call to set() in line 229 and 230, like u do in probabilistic.py

Reply:
what do you mean exactly? Trying to solve it